programs/svm-spoke/src/state/refund_account.rs

```diff
@@ -0,0 +1,132 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token_interface::TokenAccount;
+
+use crate::error::CustomError;
+
+#[account]
+#[derive(InitSpace)]
+pub struct ClaimAccount {
+    pub amount: u64,
+    pub initializer: Pubkey,
+}
+
+// When executing relayer refund leaf, refund accounts are passed as remaining accounts and can hold either a regular
+// token account or a claim account. This enum is used to differentiate between the two types.
+pub enum RefundAccount<'info> {
+    TokenAccount(InterfaceAccount<'info, TokenAccount>),
+    ClaimAccount(Account<'info, ClaimAccount>),
+}
+
+impl<'c, 'info> RefundAccount<'info>
+where
+    'c: 'info,
+{
+    // This function is used to parse a refund account from the remaining accounts list. It first tries to parse it as
+    // a token account and if that fails, it tries to parse it as a claim account.
+    pub fn try_from_remaining_account(
+        remaining_accounts: &'c [AccountInfo<'info>],
+        index: usize,
+        expected_token_account: &Pubkey,
+        expected_mint: &Pubkey,
+        token_program: &Pubkey,
+    ) -> Result<Self> {
+        let refund_account_info = remaining_accounts
+            .get(index)
+            .ok_or(ErrorCode::AccountNotEnoughKeys)?;
+
+        Self::try_token_account_from_account_info(
+            refund_account_info,
+            expected_token_account,
+            expected_mint,
+            token_program,
+        )
+        .map(Self::TokenAccount)
+        .or_else(|| {
+            Self::try_claim_account_from_account_info(
+                refund_account_info,
+                expected_mint,
+                expected_token_account,
+            )
+            .map(Self::ClaimAccount)
+        })
+        .ok_or_else(|| {
+            error::Error::from(CustomError::InvalidRefund)
+                .with_account_name(&format!("remaining_accounts[{}]", index))
+        })
+    }
+
+    // This implements the following Anchor account constraints when parsing remaining account as a token account:
+    // #[account(
+    //     mut,
+    //     address = expected_token_account @ CustomError::InvalidRefund,
+    //     token::mint = expected_mint,
+    //     token::token_program = token_program
+    // )]
+    // pub token_account: InterfaceAccount<'info, TokenAccount>,
```

Comment:
is this mean to be commented out?

Reply:
yes, left them for auditing purposes as we reimplement what Anchor macro would have done on static account.